### ISSUE TITLE
Deadline exceeded on make test

### ISSUE THREAD

**Issue by schoentoon** [golangci/golangci-lint#308]
Thank you for creating the issue!

Please include the following information:
1. Version of golangci-lint: `golangci-lint --version` 4be47944061371adb8d89081fe3d7a7a9cde526e
2. Config file: `cat .golangci.yml`
3. Go environment: `go version && go env`
```go version go1.11.2 darwin/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/<snip>/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/<snip>/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/<snip>/git/golangci-lint/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/qk/76c0lh7166nbz10bntlqwqw80000gn/T/go-build167844520=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
4. Verbose output of running: `golangci-lint run -v`

When running `make test` on said commit on my machine I'm getting a deadline exceeded. Ideally this test would actually work before I make changes so I don't depend on the CI to verify all test. Could be related to #293

```
$ make test
go build -o golangci-lint ./cmd/golangci-lint
GL_TEST_RUN=1 ./golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/toonschoenmakers/git/golangci-lint /Users/toonschoenmakers/git /Users/toonschoenmakers /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO Gocritic enabled checks: [appendAssign assignOp caseOrder dupArg dupBranchBody dupCase flagDeref ifElseChain regexpMust singleCaseSwitch sloppyLen switchTrue typeSwitchVar underef unlambda unslice defaultCaseOrder]
INFO [lintersdb] Active 24 linters: [deadcode depguard dupl errcheck gochecknoinits goconst gocritic gocyclo gofmt goimports golint govet ineffassign interfacer lll megacheck misspell nakedret scopelint structcheck typecheck unconvert unparam varcheck]
INFO [loader] Go packages loading at mode load deps types and syntax took 4.79560559s
INFO [loader] SSA repr building timing: packages building 184.364141ms, total 2.952123574s
INFO [loader] SSA for megacheck repr building timing: packages building 61.801546ms, total 2.80607197s
INFO [runner] worker.1 took 10.689395345s with stages: unparam: 6.569075531s, interfacer: 1.586052827s, gofmt: 864.686658ms, dupl: 670.149831ms, ineffassign: 503.170685ms, misspell: 323.7067ms, lll: 58.58557ms, varcheck: 43.801528ms, deadcode: 17.545629ms, errcheck: 14.092552ms, unconvert: 12.073119ms, structcheck: 11.8814ms, gocyclo: 4.885859ms, goconst: 3.6376ms, nakedret: 2.880067ms, scopelint: 2.676574ms, gochecknoinits: 96.828µs, typecheck: 6.471µs
INFO [runner] worker.3 took 18.744751966s with stages: megacheck: 18.744703338s
INFO Memory: 513 samples, avg is 1641.7MB, max is 1891.3MB
INFO Execution took 1m0.008441405s
INFO [runner] worker.2 took 1m51.5236163s with stages: goimports: 1m51.405344191s, govet: 118.199784ms
INFO [runner] worker.4 took 1m56.788360149s with stages: depguard: 1m52.277140257s, golint: 3.814028743s, gocritic: 697.050004ms
INFO [runner] Workers idle times: #1: 1m46.098746862s, #2: 5.264484623s, #3: 1m38.043573017s
INFO [runner] processing took 481.70038ms with stages: cgo: 326.76279ms, autogenerated_exclude: 72.950971ms, exclude: 49.855096ms, skip_dirs: 24.862423ms, nolint: 3.98448ms, path_prettifier: 3.259587ms, diff: 4.059µs, uniq_by_line: 3.76µs, path_shortener: 3.46µs, max_same_issues: 3.238µs, source_code: 3.138µs, skip_files: 2.701µs, max_from_linter: 2.47µs, max_per_file_from_linter: 2.207µs
ERRO Deadline exceeded: try increase it by passing --deadline option
make: *** [test] Error 4
```

**Comment by jirfag** [comment]
hi!
can you attach `.golangci.yml`?

**Comment by jitsuineccles** [comment]
Try disabling depguard

**Comment by jirfag** [comment]
may be related to https://github.com/OpenPeeDeeP/depguard/issues/7

**Comment by tpounds** [comment]
Resolving as fixed by #589 which includes https://github.com/OpenPeeDeeP/depguard/issues/7.
